Release new 1.10

marvin-bitterlich opened the pull request:
Test please

marchermans committed:
Patch in the CurseForge upload module. (#881)

I forgot to add the module when I Marvins pr.

marvin-bitterlich committed:
Feature/happiness n hunger (#880)

* Fix exceptions with blockFire

Also fix problem with coarse dirt and podzol

* Improve flint&Steel handling

And fix coarse dirt for good

* Fix the requesting of the flint and steel and fishing rod if used

* sonar fixes and more

* improve grass handling

* fix small bug

* some more sonar fixes

* Removing dependencies of Abstract class to siblings

* Fix up javadoc

* First part saturation and colony happiness

* Handle food gathering

* Some adaptions for the dman

* Adapt class check to predicate

* Refactor building code

* add missing handlers

* Improve saturation handling

* fix grasspath issue

* improve handling part 1

* fix general handler

* part 2

* fix placement handler bug

* only get certain amount

* fix bug

* Remove surpress

* suppress for now

* Update Colony.java

* Request saturation correctly

* Add requesting, fix builder bugs

* Fix logik error

* small adaptions

* Also citizens without home can talk

* Improve counter attack of guards and update the colony

* fix error with stuck dman

* fix delivery

* avoid friendly fire for counter attack

* Higher food necessity at higher building level

* Add comments and let dman take extra food with him if saturation low

* fix nullpointer exception

* Patch in the CurseForge upload module.

I forgot to add the module when I Marvins pr.

nojo121 committed:
-If the tree is a slime tree is now stored in NBT (#882)

Raycoms committed:
Feature/happiness n hunger (#887)

* Fix exceptions with blockFire

Also fix problem with coarse dirt and podzol

* Improve flint&Steel handling

And fix coarse dirt for good

* Fix the requesting of the flint and steel and fishing rod if used

* sonar fixes and more

* improve grass handling

* fix small bug

* some more sonar fixes

* Removing dependencies of Abstract class to siblings

* Fix up javadoc

* First part saturation and colony happiness

* Handle food gathering

* Some adaptions for the dman

* Adapt class check to predicate

* Refactor building code

* add missing handlers

* Improve saturation handling

* fix grasspath issue

* improve handling part 1

* fix general handler

* part 2

* fix placement handler bug

* only get certain amount

* fix bug

* Remove surpress

* suppress for now

* Update Colony.java

* Request saturation correctly

* Add requesting, fix builder bugs

* Fix logik error

* small adaptions

* Also citizens without home can talk

* Improve counter attack of guards and update the colony

* fix error with stuck dman

* fix delivery

* avoid friendly fire for counter attack

* Higher food necessity at higher building level

* Add comments and let dman take extra food with him if saturation low

* fix nullpointer exception

* Patch in the CurseForge upload module.

I forgot to add the module when I Marvins pr.

marvin-bitterlich committed:
Merge branch 'release/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'testing/1.10' into version/1.10

marvin-bitterlich committed:
Feature/buildtool server side nbt develop (#879)

* Feature/buildtool server side nbt (#777)

* Add a persistent UUID to the server, saved in minecolomies.dat
Send the UUID to the client when the player log in.
The UUID will be use to create a directory on the client to store data specific to a server

* request as schematic from the client and send the schematic back by the server

* Allow to override huts and decorations by nbt file in a directory

* Add custom decoration (from local player)
Cache file using the md5 hash as a file name inside cache directory
When the player want to use custom, he will need to send the schematic unless the cache/<md5 hash> already exists

Schematic can be send to the server only if enabled in the configuration (disable by default), no schematic management is done yet, no limitation on the number of schamatic send, unused one are still there.

* Remember the last structure build, so we can quicky build it again, usefull for walls
Store the name instead of index as they may change

* Allow to rename/delete custom schematic

Conflicts:
	src/main/java/com/minecolonies/coremod/MineColonies.java
	src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java
	src/main/java/com/minecolonies/coremod/colony/Structures.java
	src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java
	src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuild.java
	src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java
	src/main/java/com/minecolonies/coremod/network/messages/BuildToolPlaceMessage.java
	src/main/java/com/minecolonies/coremod/network/messages/SaveScanMessage.java
	src/main/java/com/minecolonies/coremod/util/ClientStructureWrapper.java

* javadoc

marvin-bitterlich committed:
Merge remote-tracking branch 'origin/version/1.10' into testing/1.10

marvin-bitterlich committed:
Update build.properties

marchermans committed:
Adding Changelog generation. (#896)

Signed-off-by: OrionOnline <oriondevelopment@outlook.com>

marvin-bitterlich committed:
Merge remote-tracking branch 'origin/version/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'release/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'release/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'testing/1.10' into version/1.10

xavierh committed:
1.10-Rewrite handleKeepX to be identical than release/1.11 (#931)

* Rewrite handleKeepX to be identical that release/1.11

* fix package info

marvin-bitterlich committed:
Merge remote-tracking branch 'origin/version/1.10' into version/1.10

marvin-bitterlich committed:
Merge branch 'release/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'testing/1.10' into version/1.10

marchermans committed:
Hotfix for the DMan (#940)

* Hotfix for the DMan not behaving properly cause it is not getting reported about the insertion result

* Fixing a type in the iteration for the extraction slots of the DMan

Raycoms committed:
Feature/limit warehouse (#948)

* should fix the no tinkers crashes

* should fix trange crash

* Fix fisherman

* limit warehouse

marvin-bitterlich committed:
Merge remote-tracking branch 'origin/release/1.10' into testing/1.10

marvin-bitterlich committed:
Merge branch 'testing/1.10' into version/1.10

Raycoms committed:
Feature/better builder lj positioning (#952)

* fix lumberjack position finding and fix builder position finding

* Fix sonar issues

marvin-bitterlich committed:
bump version

marvin-bitterlich committed:
bump version

MinecoloniesSonar commented:
SonarQube analysis reported 119 issues
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 54 minor
* ![INFO][INFO] 12 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.233 lines, which is greater than 750 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1273](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1273): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1275](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1275): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/203ba8eda13ad6920ced5fd7d870792678f901f8/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Raycoms committed:
Fix/no tarzan (#974)

* fix lumberjack position finding and fix builder position finding

* Fix sonar issues

* Fix Tarzan climbing the trees

Uhm I meant the lumberjack

xavierh committed:
remove unimplemented huts (#973)

marvin-bitterlich committed:
Merge remote-tracking branch 'origin/testing/1.10' into testing/1.10

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -74,7 +74,7 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockHutDeliveryman, 1), "XXX", "X#X", "XXX", 'X', PLANK_WOOD, '#', Items.LEATHER_BOOTS));
 
-        enableInDevelopmentFeatures(enableInDevelopmentFeatures);
+        //enableInDevelopmentFeatures(enableInDevelopmentFeatures);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

MinecoloniesSonar commented:
SonarQube analysis reported 119 issues
* ![MAJOR][MAJOR] 52 major
* ![MINOR][MINOR] 55 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.233 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1273](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1273): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1275](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1275): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/4d5761dfde875a45bf3b924b42a972185536c047/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Raycoms committed:
Hotfix/low stats (#976)

* fix lumberjack position finding and fix builder position finding

* Fix sonar issues

* Fix Tarzan climbing the trees

Uhm I meant the lumberjack

* fix bug where citizen start all up with 1

WORMSS committed:
Average Housing on Happiness Calculation (#981)

Fixes: #978

marvin-bitterlich committed:
Merge branch 'version/1.10' into testing/1.10

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyView.java:
```diff
@@ -59,7 +64,7 @@
     private Set<Block> freeBlocks = new HashSet<>();
 
-    private double overallHappiness;
+    private double overallHappiness = 5;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented:
SonarQube analysis reported 120 issues
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 55 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.233 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1273](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1273): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1275](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1275): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/29a5d48c56000dc7933176b37c15571472b897da/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

xavierh committed:
1.10 d-man delivery fix (#986)

* 1.11 d-man delivery fix (#980)

* Fix forceTransferStack return value
Fix some java doc

* Fix javadoc

* Fix happiness calculation about housing

* Transfer for real instead of simulating

* remove some warnings

Conflicts:
	src/main/java/com/minecolonies/coremod/colony/Colony.java
	src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

* sonar

Raycoms committed:
Fix/builder duping 10 (#988)

* fix lumberjack position finding and fix builder position finding

* Fix sonar issues

* Fix Tarzan climbing the trees

Uhm I meant the lumberjack

* fix bug where citizen start all up with 1

* Fix problem with builder requesting double

* don't request solid placeholder if world is solid

* make citizen only tp if employed

* recalculate after reload

* balance happiness depending on guards

Raycoms committed:
Fix/colony crash10 (#998)

* Stop crashing with other world events

* add new creativetab design to 1.10

marvin-bitterlich committed:
Sonar/warnings (#992) (#995)

* Sonar/warnings (#992)

* Fix Blockers

* first draft for translation constants

* fix formatting

* fix some constants and other issues

* formatting

* fix some ternaries

* sonar fixes

* add suppress

* add doc files

* reorder

* wtf java

* last fixes

* remove unused

* Update InventoryCitizen.java

* Update WindowHireWorker.java

* Update WindowHireWorker.java

Raycoms committed:
Feature/builder pickup (#999)

* Make builder pickup items after building

* sonar fix cast to double

* remove unused import

Raycoms committed:
Add female dman sounds + functionality for hostile and saturation (#990)

* Add female dman sounds + functionality for hostile and saturation

* be happy sonar

* fix nullpointer exception

* increase chance

* fix small bug

* remove import

* Update EntityAICitizenAvoidEntity.java

* Fix dman sound surpress template

* stage

* Might look better like this.

* add package info

* Fix problematic behavior

* imports

Raycoms committed:
Feature/builder pickup (#1002)

* Make builder pickup items after building

* sonar fix cast to double

* remove unused import

* remove double guard code 1.10

* unused import

* Update doc

xavierh committed:
Feature/1.10 better chat (#994) (#1007)

* Feature/1.10 better chat (#994)

* Suppress some warning

* Add worker name to the message when chest is full

* remove warnings

* Builder say when then start/end a job

* Shorter chat version for colony owner

* D-man only say that the chest if full when ihe cannot swap
D-man say whose chest is full

* Change dman saying

* Compare itemstack properly

* Different message when pick axe requested level is greater than hut level

* D-man hold the item he will deliver or the item he did gather
Fix javadoc

* EntityCitizen does not know about BuildingFarmer now

* remove most of the Jobs from EnityCitizen

* dman hold the correct item

Conflicts:
	src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java
	src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java
	src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java
	src/main/resources/assets/minecolonies/lang/en_US.lang

* add some package info

* package info

marvin-bitterlich committed:
Merge branch 'version/1.10' into testing/1.10

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java:
```diff
@@ -57,10 +58,10 @@ public BlockBarrel()
     }
 
     //todo: register block with new method
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java:
```diff
@@ -220,7 +220,7 @@ public Object handle(@NotNull final World world, @NotNull final BlockPos pos, @N
     public static class FlowerPotPlacementHandler implements IPlacementHandler
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 31 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented:
SonarQube analysis reported 171 issues
* ![MAJOR][MAJOR] 81 major
* ![MINOR][MINOR] 69 minor
* ![INFO][INFO] 21 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.231 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1163](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1163): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1164](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1164): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1276](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1276): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1278](https://github.com/Minecolonies/minecolonies/blob/5c2c80e86d541d42ccfc1d6ae23344c55c4ee36e/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1278): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/sounds/ModSoundEvents.java:
```diff
@@ -5,12 +5,20 @@
     /**
+     * List of sound handlers.
+     */
+    public static final List<AbstractWorkerSounds> handlers = new ArrayList<>();
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this member "protected". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2386)

Raycoms committed:
make waypoints render when buildtool open (#1004)

* make waypoints render when buildtool open

* Fix javadoc

* javadoc

* Add colony border rendering as well

* fix sonar issues

* fix door duping of builder

* Make marvin happy, add package info.

* fix up buggy behavior

* fix annotations and sonar issues

* fix it up

xavierh committed:
Refactor/1.11 workorders (#1012) (#1013)

* Suppress some warning

* Add worker name to the message when chest is full

* Fix warnings for PlacementHandlers

* Builder say when then start a job

* WorkOrderBuildHut extends WorkOrderBuildDecoration
instead of
WorkOrderBuildDecoration extends WorkOrderBuild

* revert WorkOrderBuildHut class to WorkOrderBuild

* Work Manager does not need to know about Construction tape and WorkOrderBuildDecoration

* JobBuilder class does not need to know about WorkOrderBuild

* remove duplicated code

* add package info

* Sonar suppress

Conflicts:
	src/main/java/com/minecolonies/coremod/colony/WorkManager.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

xavierh committed:
bug #1014 fix crash when a colony does not have any owner (#1015)

* fix crash when a colony does not have any owner

* remove logging

MinecoloniesSonar commented:
SonarQube analysis reported 171 issues
* ![MAJOR][MAJOR] 81 major
* ![MINOR][MINOR] 69 minor
* ![INFO][INFO] 21 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.231 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1163](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1163): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1164](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1164): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1276](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1276): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1278](https://github.com/Minecolonies/minecolonies/blob/d54d1890139883a717c43c37561455c139dd4a7c/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1278): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

marvin-bitterlich committed:
Merge branch 'version/1.10' into testing/1.10

cltnschlosser committed:
Bugfix/#758 concurrency issue (#1025) (#1029)

* Centralize the copying of the needed items list, and make it explicitly clear a copy is happening

* Remove duplicated methods, and unnecessary casts

* Change CopyOnWriteArrayList to ConcurrentLinkedQueue for current warehouse tasks.

The list is being modified often, and being treated like a queue. The ConcurrentLinkedQueue should be a better data structure for the situation and be more performant.

cltnschlosser committed:
Check if citizen is jobless before triggering job death achievement (#1038)

cltnschlosser committed:
Dismount before teleporting, and add check for invalid line width (#1036)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java:
```diff
@@ -107,30 +105,16 @@ public static void placeConstructionTape(@NotNull AbstractBuilding building, @No
      */
 
-    public static void placeConstructionTape(@NotNull WorkOrderBuild workOrder, @NotNull World world)
+    public static void placeConstructionTape(@NotNull WorkOrderBuildDecoration workOrder, @NotNull World world)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java:
```diff
@@ -20,10 +19,8 @@
 import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
 import java.util.List;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.util.List'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/ColonyView.java:
```diff
@@ -59,7 +64,15 @@
+    /**
+     * The overall happiness of the colony.
+     */
+    private double overallHappiness = 5;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java:
```diff
@@ -37,17 +68,292 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+     * @param citizen which could build it or not
+     * @return true if he is able to.
+     */
+    @SuppressWarnings("squid:S1172")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1172' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java:
```diff
@@ -37,17 +68,292 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+     * @param hasBuilder true if we have a builder for this work order
+     * @param sendMessage true if we need to send the message
+     */
+    @SuppressWarnings("squid:S1172")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1172' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuildDecoration.java:
```diff
@@ -37,17 +68,292 @@ public WorkOrderBuildDecoration(final String structureName, final String workOrd
+     * @param world where the decoration is
+     * @return building rotation.
+     */
+    @SuppressWarnings("squid:S1172")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1172' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/workorders/AbstractWorkOrder.java:
```diff
@@ -285,9 +285,13 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
      * @param colony The colony that owns the Work Order
      * @return True if the WorkOrder is still valid, or False if it should be deleted
      */
+    @SuppressWarnings("squid:S1172")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1172' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:
```diff
@@ -92,10 +93,10 @@ public void update()
                     if(buildingEntry.getValue() instanceof AbstractBuildingWorker
-                            && !list.contains(buildingEntry.getValue())
-                            && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                            && !taskQueue.contains(buildingEntry.getValue())
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:
```diff
@@ -187,13 +187,14 @@ public boolean checkInWareHouse(@NotNull final AbstractBuilding buildingEntry, b
             }
-            if (list.contains(buildingEntry))
+
+            if (taskQueue.contains(buildingEntry))
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:
```diff
@@ -117,32 +118,28 @@ public boolean checkInWareHouseForFood(final BuildingHome buildingEntry, final b
             }
 
-            if (list.contains(buildingEntry))
+            if (taskQueue.contains(buildingEntry))
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:
```diff
@@ -167,14 +166,15 @@ public boolean checkInWareHouse(@NotNull final AbstractBuilding buildingEntry, b
             }
-            if (list.contains(buildingEntry))
+
+            if (taskQueue.contains(buildingEntry))
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented:
SonarQube analysis reported 185 issues
* ![MAJOR][MAJOR] 94 major
* ![MINOR][MINOR] 62 minor
* ![INFO][INFO] 29 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.237 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1163](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1163): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1164](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1164): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1279](https://github.com/Minecolonies/minecolonies/blob/a425532d8af28106ed05328772cc9abfb7d3671b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1279): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

xavierh committed:
Feature/1.10 efficient dman (#1044)

* Feature/1.10 efficient dman (#1017)

* gather from the chest which have just been delivered

* Deliver mor ethan one stack at a time depending of the deliveryman hut's level

* * in BuildingBuilder override neededForWorker instead of using the custom requiresResourceForBuilding method
* Add container to building event when the builder did not have to build it

* register block in AbstractBuilding instead of adding container only, so that extending class could register any type of block, not only chests

* remove checkInWareHouseForFood method in favour of polymorphisme

* remove duplicated code

* javadoc

* language constant

* revert back some changes

Conflicts:
	src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java
	src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java

* Citizen hut need food

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:
```diff
@@ -92,14 +95,10 @@ public void update()
-                        checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue(), true);
-                    }
-                    else if(buildingEntry.getValue() instanceof BuildingHome && ((BuildingHome) buildingEntry.getValue()).isFoodNeeded())
+                            && !taskQueue.contains(buildingEntry.getValue())
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented:
SonarQube analysis reported 190 issues
* ![MAJOR][MAJOR] 95 major
* ![MINOR][MINOR] 66 minor
* ![INFO][INFO] 29 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowHireWorker.java#L125](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/client/gui/WindowHireWorker.java#L125): Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowHutFarmer.java#L206](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/client/gui/WindowHutFarmer.java#L206): Reduce this anonymous class number of lines from 37 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.237 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 83 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1163](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1163): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1164](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1164): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1277](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1277): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1279](https://github.com/Minecolonies/minecolonies/blob/4a09daefeee8c0fe46a43753bc830cf026a80a11/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1279): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
